modules/livestore/instance.go

```diff
@@ -334,3 +338,89 @@ func (i *instance) deleteOldBlocks() error {
 
 	return nil
 }
+
+func (i *instance) FindByTraceID(ctx context.Context, traceID []byte) (*tempopb.TraceByIDResponse, error) {
+	i.blocksMtx.RLock()
+	defer i.blocksMtx.RUnlock()
+
+	var completeTrace *tempopb.Trace
+	metrics := tempopb.TraceByIDMetrics{}
+	combiner := trace.NewCombiner(math.MaxInt64, true)
+	_, err := combiner.Consume(completeTrace)
+	if err != nil {
+		return nil, err
+	}
+	// TODO MRD this code looks ripe for simplification. Its the same pattern repeated several times.
+	i.liveTracesMtx.Lock()
+	if liveTrace, ok := i.liveTraces.Traces[util.HashForTraceID(traceID)]; ok {
+
+		tempTrace := &tempopb.Trace{}
+		tempTrace.ResourceSpans = liveTrace.Batches
+		// Previously there was some logic here to add inspected bytes in the ingester. But its hard to do with the different
+		// live traces format and feels inaccurate.
+		_, err := combiner.Consume(tempTrace)
+		if err != nil {
+			i.liveTracesMtx.Unlock()
+			return nil, fmt.Errorf("unable to unmarshal liveTrace: %w", err)
+		}
+	}
+	i.liveTracesMtx.Unlock()
+
+	// TODO MRD Add limits
+
+	// Loop over all the complete blocks looking for a specific ID. The implementation looks like it will return nil if the trace is not found.
```

Comment:
why shouldn't we trust nil = not found?

Reply:
I can remove the comment, I am not checking for it thoroughly enough like I previously was. Combiner will handle if `trace.Trace` is nil.